app/javascript/controllers/wunderbaum_controller.js

```diff
@@ -17,6 +17,14 @@ export default class extends Controller {
 
   async connect() {
     try {
+      // Fetch options first, before creating the tree
+      await Promise.all([
```

Comment:
We are really trying to avoid adding things like this in.  When you add promises in before we even start to initialize the tree it is going to make performance even slower than it already is. The data is only used in dropdown menus, so it can wait until after the tree is visible.

Reply:
It doesn't populate the select lists in that case, which is why I moved it up.